### PR TITLE
Insert an object with an array column (close #3018)

### LINKED ISSUE #3018: Document workaround for array input
Until the time [this](https://github.com/hasura/graphql-engine/issues/348) is implemented, we should document how to pass array as literals.

Could be a section after [this](https://docs.hasura.io/1.0/graphql/manual/mutations/insert.html#insert-an-object-with-a-jsonb-column): called "Insert an object with an array column".

### PR DESCRIPTION
**Pull request by larainfo**
Insert an object with an array column
fix #3018 

### Description

Example of an insert an object with an array column.

### Affected components 

- Docs

### Related Issues

#3018 
#2243

### CODE REVIEW

**Comment by marionschleifer** [comment]
@larainfo thank you for your contribution 🙂 we are looking for an example that passes an array as a literal. 

The column type should be `text[]` and the input value has to be `"{hello,world}"`, as described here: https://github.com/hasura/graphql-engine/issues/348.

Could you implement the changes?

**Comment by marionschleifer** [comment]
@larainfo are you interested in making the necessary changes? We have another PR open on this issue waiting for a review.

**Comment by larainfo** [comment]
> @larainfo are you interested in making the necessary changes? We have another PR open on this issue waiting for a review.

@marionschleifer I did the example as sugested by [https://github.com/hasura/graphql-engine/issues/348#issuecomment-460017748] . If you want it to be another way, please tell me.

**Comment by marionschleifer** [comment]
@larainfo I don't see any changes. Are you sure you pushed them?

**Comment by marionschleifer** [comment]
@larainfo any update here? Also, could you please send an email to marion@hasura.io, so that we can arrange the shipping of your swag? 🙂

**Comment by larainfo** [comment]
> @larainfo any update here? Also, could you please send an email to [marion@hasura.io](mailto:marion@hasura.io), so that we can arrange the shipping of your swag? 🙂

@marionschleifer can you check my latest commit, please?

File: docs/graphql/manual/mutations/insert.rst [hunk 1]
```diff
@@ -371,6 +371,91 @@ Insert an object with a JSONB column
       }
     }
 
+Insert an object with an array column
+------------------------------------
```
Comment (marionschleifer) [inline docs/graphql/manual/mutations/insert.rst]:
```suggestion
-------------------------------------
```

File: docs/graphql/manual/mutations/insert.rst [hunk 2]
```diff
@@ -371,6 +371,91 @@ Insert an object with a JSONB column
...
+If you need to evaluate an array field you can use ``jsonb`` or ``literal strings``.
```
Comment (marionschleifer) [inline docs/graphql/manual/mutations/insert.rst]:
```suggestion
The way to insert an array field depends on its type. If the type is JSONB, you can pass a JSONB array as a variable. If it's a Postgres array, you have to pass the array as a string literal.
```

File: docs/graphql/manual/mutations/insert.rst [hunk 3]
```diff
@@ -371,6 +371,91 @@ Insert an object with a JSONB column
...
+
+**Example 1:** Insert a new ``article`` object with a array ``tags`` column using jsonb:
```
Comment (marionschleifer) [inline docs/graphql/manual/mutations/insert.rst]:
```suggestion
**Example 1:** Insert a new ``article`` object with an array ``tags`` column using JSONB:
```

File: docs/graphql/manual/mutations/insert.rst [hunk 4]
```diff
@@ -371,6 +371,91 @@ Insert an object with a JSONB column
...
+          ]
+        }
+      }
+    }
+  :variables:
+    {
+      "tags": ["tag1", "tag2"]
+    }
+    
+**Example 2:** Insert a new ``article`` passing tags as literal strings:
```
Comment (marionschleifer) [inline docs/graphql/manual/mutations/insert.rst]:
```suggestion
**Example 2:** Insert a new ``article`` passing tags as literal strings:

```

Comment (marionschleifer) [reply docs/graphql/manual/mutations/insert.rst]: There's a white space missing which breaks the display of the example:

image

**Comment by rikinsk** [comment]
@marionschleifer made a couple of changes.

- removed the jsonb note as that should be clear form the previous jsonb section
- added link to postgres array literals docs
- changes the example to not use tags as we use tags for many to many relationships and users might confuse this with that
- made the example with and and without variables

**Pull request merged**